packages/feed-list/src/index.ts

```diff
@@ -0,0 +1,226 @@
+import { Readable } from 'stream'
+import {
+  FEED_MAX_DATA_LENGTH,
+  BaseBzz,
+  BaseResponse,
+  FeedID,
+  FeedParams,
+  FetchOptions,
+} from '@erebos/api-bzz-base'
+import { createHex, Hex, hexInput } from '@erebos/hex'
+
+export const MAX_CHUNK_BYTE_LENGTH = FEED_MAX_DATA_LENGTH
+// Hex value is a hex string (= byte length x 2) prefixed by `0x`
+export const MAX_CHUNK_VALUE_LENGTH = MAX_CHUNK_BYTE_LENGTH * 2 + 2
+
+export interface ForwardsChunkIterator<T> extends AsyncIterator<T> {
+  length: number
+}
+
+export interface ChunkListReaderConfig<
+  Bzz extends BaseBzz<BaseResponse, Readable> = BaseBzz<BaseResponse, Readable>
+> {
+  bzz: Bzz
+  feed: FeedID | FeedParams
+  fetchOptions?: FetchOptions
+}
+
+export class ChunkListReader<
+  Bzz extends BaseBzz<BaseResponse, Readable> = BaseBzz<BaseResponse, Readable>
+> {
+  public bzz: Bzz
+  public fetchOptions: FetchOptions
+  protected id: FeedID
+
+  constructor(config: ChunkListReaderConfig<Bzz>) {
+    this.bzz = config.bzz
+    this.fetchOptions = config.fetchOptions || {}
+    this.id =
+      config.feed instanceof FeedID ? config.feed : new FeedID(config.feed)
+  }
+
+  public async load(index: number): Promise<Hex | null> {
+    const id = this.id.clone()
+    id.time = index
```

Comment:
Maybe the `id` should be only set after the the data is read, otherwise this can leave the object in an invalid state.

Reply:
Yes that's why the `id` is cloned, so it doesn't affect the one on the instance.

Reply:
Ah ok, sorry :)